raven/src/main/java/com/getsentry/raven/connection/AbstractConnection.java

```diff
@@ -78,6 +86,9 @@ public final void send(Event event) {
             doSend(event);
             waitingTime = baseWaitingTime;
         } catch (ConnectionException e) {
+            for (EventSendFailureCallback eventSendFailureCallback : eventSendFailureCallbacks) {
+                eventSendFailureCallback.onFailure(event, e);
```

Comment:
Yeah, probably so. But then in the `catch` I think we need to avoid using any logging frameworks (as do the callbacks themselves)... I'm not sure how we (and the callbacks) can report errors to the user. :(